### DISCUSSION TITLE
Clarification on DataDog Observability Metrics

### DISCUSSION THREAD

**Discussion by rimbesat** [DataDog/chaos-controller#579] [Q&A]
Our Dashboard currently does show the majority of our experiments. However, on a few experiments the chaos.controller.disruptions.count displays a value below 1. 
For example:
<img width="1746" alt="image" src="https://user-images.githubusercontent.com/93604675/186432209-4405eb37-be49-4c10-8da0-4a8e9bcb28dd.png">
In most of these cases the experiment has a count metric of 0.1.
<img width="1743" alt="image" src="https://user-images.githubusercontent.com/93604675/186432763-4e3dc11f-8e43-42ea-b175-59e9559408af.png">

We would like to know if you have come across this issue before, or any ideas of what has gone wrong?

**Comment by ptnapoleon** [comment]
These screenshots are from a DataDog dashboard? The metrics themselves are hopefully being emitted correctly, but they might be rolled up incorrectly

**Reply by chawlabhishek** [reply]
@ptnapoleon Thanks for the quick response. We had a couple of questions based on your response.

- Could you please confirm a way to validate if the metrics are being emitted correctly? 
- Also, Could you provide more context on metrics getting rolled up incorrectly?

Thanks!

**Reply by ptnapoleon** [reply]
> Could you please confirm a way to validate if the metrics are being emitted correctly?

I don't know a way other than using DataDog to confirm we're seeing metrics when expected. 

> Also, Could you provide more context on metrics getting rolled up incorrectly?

If the metrics are being evaluated as a `rate` rather than a `count` on the query side, it would result in these small floating points

**Reply by rimbesat** [reply]
Alright thanks,
Is there an option to have a metrics endpoint for the controller so we can see metrics, or its only done via the push method, dogstatsd?

**Reply by ptnapoleon** [reply]
@Devatoria Do you know? ^ I do not

**Reply by Devatoria** [reply]
Excuse me for the long delay in response. As of today we can't do that, we're using the Datadog statsd library directly. However, the [metrics package](https://github.com/DataDog/chaos-controller/tree/main/metrics) uses and interface so we can implement other ways to expose metrics. We can imagine implementing another way (prometheus for instance) of exposing metrics. It is not that complex as it is just implementing the interface.

**Reply by nikos912000** [reply]
Hi @ptnapoleon @Devatoria. Thanks for responding.
We have filed a DataDog ticket for this. After spending some time on it, this might have to do with a problematic cluster that causes the decimals. I hope the agent flare will give us more context.
